lib/internal/modules/cjs/loader.js

```diff
@@ -829,9 +829,19 @@ Module.runMain = function() {
 };
 
 Module.createRequireFromPath = (filename) => {
-  const m = new Module(filename);
-  m.filename = filename;
-  m.paths = Module._nodeModulePaths(path.dirname(filename));
+  // Allow a directory to be passed as the filename
+  const normalized = path.win32.normalize(filename);
```

Comment:
I've not seen Node core code that normalizes on `win32` (for all) for a backslash check. It just makes me uneasy to use a hardcoded `win32` route since for me it's the route bugs tend to slip into.

Reply:
@jdalton another suggestion for this?

Reply:
Agreed it would be better not to rely on Windows path resolution in Linux to make this work. 

The logic it's trying to capture is a trailing `/` on Linux, and a trailing `/` or `\\` on Windows. So why not just be explicit:

```suggestion
const trailingSlash = filename.endsWith(path.sep) || path.sep !== '/' && filename.endsWith('\\');
```

Reply:
Fixed